src/sparseml/optim/helpers.py

```diff
@@ -512,3 +518,118 @@ def _maybe_parse_number(val: str) -> Union[str, float, int]:
             return float(val)
         except Exception:
             return val
+
+
+def _extract_metadata_from_recipe(container):
+    metadata = {}
+    if RECIPE_METADATA_KEY in container.keys():
+        metadata = container[RECIPE_METADATA_KEY]
+    return metadata
+
+
+def _extract_metadata_from_staged_recipe(container):
+    metadata = {}
+    stage_names = _get_recipe_stage_names(container)
+    for stage_name in stage_names:
+        if RECIPE_METADATA_KEY in container[stage_name].keys():
+            metadata[stage_name] = container[stage_name][RECIPE_METADATA_KEY]
+    if metadata and (len(metadata) != len(stage_names)):
+        raise ValueError(
+            "It seems that some stages in your checkpoint recipe"
+            "contain metadata and some don't."
+            "This is not allowed: either all the "
+            f"stages contain {RECIPE_METADATA_KEY} key or none does."
```

Comment:
```suggestion
            "Some stages in recipe do not. Either all or no stages must "
            f"must contain the {RECIPE_METADATA_KEY} key"
```

editing for brevity and spacing